packages/esm-patient-chart-app/src/clinical-views/utils/helpers.ts

```diff
@@ -0,0 +1,188 @@
+import { launchPatientWorkspace } from '@openmrs/esm-patient-common-lib';
+import { type FormSchema } from '@openmrs/esm-form-engine-lib';
+import { formatDate, parseDate, formatDatetime } from '@openmrs/esm-framework';
+import { fetchPatientRelationships } from '../../encounter-list/encounter-list.resource';
+
+type LaunchAction = 'add' | 'view' | 'edit' | 'embedded-view';
+
+export function launchEncounterForm(
+  form: FormSchema,
+  action: LaunchAction = 'add',
+  onFormSave: () => void,
+  title?: string,
+  encounterUuid?: string,
+  intent: string = '*',
+  workspaceWindowSize?: 'minimized' | 'maximized',
+  patientUuid?: string,
+) {
+  launchPatientWorkspace('patient-form-entry-workspace', {
+    workspaceTitle: form.name,
+    mutateForm: onFormSave,
+    formInfo: {
+      encounterUuid,
+      formUuid: form.name,
+      patientUuid: patientUuid,
+      visitTypeUuid: '',
+      visitUuid: '',
+      visitStartDatetime: '',
+      visitStopDatetime: '',
+      additionalProps: {
+        mode: action === 'add' ? 'enter' : action,
+        formSessionIntent: intent,
```

Comment:
```suggestion
        formSessionIntent: intent,
        openClinicalFormsWorkspaceOnFormClose: false,
```

Reply:
Add this flag to prevent auto opening of the clinical forms workspace when a form is cancelled.